aws_lambda_powertools/event_handler/openapi/encoders.py

```diff
@@ -0,0 +1,346 @@
+import dataclasses
+import datetime
+from collections import defaultdict, deque
+from decimal import Decimal
+from enum import Enum
+from pathlib import Path, PurePath
+from re import Pattern
+from types import GeneratorType
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+from uuid import UUID
+
+from pydantic import BaseModel
+from pydantic.color import Color
+from pydantic.types import SecretBytes, SecretStr
+
+from aws_lambda_powertools.event_handler.openapi.compat import _model_dump
+from aws_lambda_powertools.event_handler.openapi.types import IncEx
+
+"""
+This module contains the encoders used by jsonable_encoder to convert Python objects to JSON serializable data types.
+"""
+
+
+def jsonable_encoder(  # noqa: PLR0911
```

Comment:
Our current json serializer is not nearly as powerful as this one. So I had to find a way to get a more powerful one inside the new OpenAPI code.